hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -81,82 +85,166 @@ public final class HoodieFileGroupReader<T> implements Closeable {
   private ClosableIterator<T> baseFileIterator;
   private HoodieRecordMerger recordMerger;
 
-  public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
-                               HoodieTableMetaClient metaClient,
-                               String fileGroupId,
-                               TypedProperties props,
-                               HoodieTimeline timeline,
-                               HoodieTableQueryType queryType,
-                               Option<String> instantTime,
-                               Option<String> startInstantTime,
-                               boolean shouldUseRecordPosition) throws Exception {
-    // This constructor is a placeholder now to allow automatically fetching the correct list of
-    // base and log files for a file group.
-    // Derive base and log files and call the corresponding constructor.
-    this(readerContext, metaClient.getHadoopConf(), metaClient.getBasePathV2().toString(),
-        instantTime.get(), Option.empty(), Option.empty(),
-        new TableSchemaResolver(metaClient).getTableAvroSchema(),
-        props, 0, Long.MAX_VALUE, shouldUseRecordPosition);
-  }
+  private final Schema dataSchema;
+  private final Schema requestedSchema;
+
+  private final Schema requiredSchema;
+
```

Comment:
What are the meanings for these two schemas?

Reply:
requested schema - the schema the caller wants (ie when next() is called, what will the schema of the record be)
required schema - the schema that we need to read from the file, for example, mor needs additional fields for merging

Reply:
does this align with the "requiredSchema" in ParquetFileFormat? Since I did not see "requestedSchema" before.

Reply:
It actually really doesn't. Requested schema is what is requiredSchema in ParquetFileFormat

Reply:
Schemas with different meanings are floating in Hudi so far. We should try to make it consistent. Please take a note, and refactor it after landing this PR. (We may also do similar things for Flink/Presto/Trino/etc later).